src/registers.rs

```diff
@@ -104,9 +108,28 @@ impl Registers {
             8 => self.pc = val,
             9 => self.cond = val,
             _ => {
-                println!("Invalid registers at set");
-                flush_stdout();
+                return Err(VmError::InvalidRegister(
+                    "Invalid registers at set".to_string(),
+                ));
             }
         }
+        Ok(())
+    }
+
+    /// Updates the condition flags in the `Registers` struct.
+    ///
+    /// # Returns
+    ///
+    /// An `Ok` result if the operation was successful, otherwise a `VmError`.
+    ///
+    pub fn update_flags(&mut self, r: u16) -> Result<(), VmError> {
+        if self.get(r)? == 0 {
+            self.cond = FL_ZRO;
+        } else if (self.get(r)? >> 15) & 1 == 1 {
+            self.cond = FL_NEG;
+        } else {
+            self.cond = FL_POS;
+        }
+        Ok(())
```

Comment:
```suggestion
    pub fn update_flags(&mut self, r: u16) -> Result<(), VmError> {
        let r_value = self.get(r)?
        if r_value == 0 {
            self.cond = FL_ZRO;
        } else if (r_value >> 15) & 1 == 1 {
            self.cond = FL_NEG;
        } else {
            self.cond = FL_POS;
        }
        Ok(())
```

I would change update_flags to look something like this. It is not important though. I'm not even sure if it looks better this way 😛

Reply:
Done! It's better this way as it uses `get` just once :)